### ISSUE TITLE
Multiplayer and C3 Runtime. High CPU load

### ISSUE THREAD

**Issue by Sowhenifoundfreenickname** [Scirra/Construct-3-bugs#2175]
## Problem description

High CPU load when you use plugin Multiplayer. Clear project, CPU ~ 1% when you use С3 Runtime or C2 Runtime. When you add Multiplayer plugin CPU ~ 7-10% when you use C3 Runtime and ~1% when you use С2 Runtime. In a normal project with objects and events, the difference in CPU load in C3 Runtime - 8-12% without Multiplayer plugin and 20-30% when you add Multiplayer plugin. This is a very big difference, which is negative at the work of the application on the smartphone.

## Attach a .c3p

[GitHub.zip](https://github.com/Scirra/Construct-3-bugs/files/2503513/GitHub.zip)

## Steps to reproduce

1. Open project
2. Start debug preview
Project #1 have Multiplayer. Project #2 without Multiplayer

## Observed result

High CPU load

## Expected result

C3 Runtime CPU load values are comparable with C2 Runtime

## Affected browsers

* Chrome: yes/no
* Firefox: yes/no
* Edge: yes/no
* Safari: yes/no

## System details

<details><summary>View details</summary>

PASTE HERE

</details>

**Comment by Sowhenifoundfreenickname** [comment]
Sorry, I used the # symbol incorrectly. Do not pay attention to the links that appear when you click on this symbol.

**Comment by AshleyScirra** [comment]
The difference is small enough that I can't measure any useful information from Chrome's profiler. I'm afraid we cannot investigate performance issues with such small discrepencies; you should try to reproduce it with a big, clear difference for us to be able to investigate it (e.g. 50% vs. 100% CPU).

I suspect this is partly due to the fact that Construct's CPU measurements are timer-based, which is inherently inaccurate since time measurements vary depending on the timer precision and the power state of the hardware (e.g. the CPU clocking down to save power will increase the time it takes to do some work, misleadingly producing a higher measurement). The CPU measurement is generally only reliable under maximum load, i.e. in projects that stress the system close to 100% CPU (as [noted in the manual](https://www.construct.net/en/make-games/manuals/construct-3/system-reference/system-expressions)). I suspect it is also partly due to the fact that parts of the Multiplayer plugin in the C3 runtime are now asynchronous, whereas the C2 runtime is synchronous. When the C3 runtime does something asynchronous, it can remain idle for a period of time until the result arrives; this will also increase timer-based measurements, which is also misleading because the runtime was actually idle and available to do more work in the intervening time.

tl;dr - timer-based CPU measurements are unreliable in several ways, and we can only investigate issues that show a large discrepancy while the system is under full load.
